spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/si/SpringIntegrationChannelBinder.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.integration.handler.BridgeHandler;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+
+/**
+ * Implementation of {@link Binder} backed by Spring Integration framework.
+ * It is useful for localized demos and testing.
+ * <p>
+ * This binder extends from the same base class ({@link AbstractMessageChannelBinder}) as
+ * other binders (i.e., Rabbit, Kafka etc). Interaction with this binder is done via a pair of named {@link MessageChannel}s.
+ * <br>
+ * The names of the channels are:
+ * <ul>
+ * <li>{@link SpringIntegrationBinderConfiguration#BINDER_INPUT}</li>
+ * <li>{@link SpringIntegrationBinderConfiguration#BINDER_OUTPUT}</li>
+ * </ul>
+ * Simply autowire them in your your application and send/receive messages.
+ * </p>
+ * You must also add {@link SpringIntegrationBinderConfiguration} to your configuration.
+ * Below is the example using Spring Boot test.
+ * <pre class="code">
+ *
+ * &#064;RunWith(SpringJUnit4ClassRunner.class)
+ * &#064;SpringBootTest(classes = {SpringIntegrationBinderConfiguration.class, TestWithSIBinder.MyProcessor.class})
+ * public class TestWithSIBinder {
+ *     &#064;Autowired
+ *     private MessageChannel BINDER_INPUT;
+ *
+ *     &#064;Autowired
+ *     private QueueChannel BINDER_OUTPUT;
+ *
+ *     &#064;Test
+ *     public void testWiring() {
+ *         BINDER_INPUT.send(new GenericMessage<String>("Hello"));
+ *         assertEquals("Hello world", new String((byte[])BINDER_OUTPUT.receive().getPayload(), StandardCharsets.UTF_8));
+ *     }
+ *
+ *     &#064;SpringBootApplication
+ *     &#064;EnableBinding(Processor.class)
+ *     public static class MyProcessor {
+ *         &#064;Transformer(inputChannel = Processor.INPUT, outputChannel = Processor.OUTPUT)
+ *         public String transform(String in) {
+ *             return in + " world";
+ *         }
+ *     }
+ * }
+ * </pre>
+ *
+ * @author Oleg Zhurakousky
+ */
+class SpringIntegrationChannelBinder extends AbstractMessageChannelBinder<ConsumerProperties,
+	ProducerProperties, SpringIntegrationProvisioner> implements DisposableBean {
+
+	SpringIntegrationChannelBinder(SpringIntegrationProvisioner provisioningProvider) {
+		super(provisioningProvider);
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		// noop
+	}
+
+	@Override
+	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
+			ProducerProperties producerProperties, MessageChannel errorChannel) throws Exception {
+		MessageChannel siBinderOutputChannel = ((SpringIntegrationProducerDestination)destination).getOutputChannel();
+		MessageHandler messageHandler = new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				siBinderOutputChannel.send(message);
```

Comment:
There is just a `BridgeHandler` for such a purpose.

Reply:
Not following you

Reply:
```
BridgeHandler handler = new BridgeHandler();
handler.setOutputChannel(siBinderOutputChannel);
return handler;
```
(Sorry didn't tested 😄 )